tests/attr/test_lime.py

```diff
@@ -0,0 +1,508 @@
+        inp = torch.tensor([[2.0, 10.0, 3.0]], requires_grad=True)
+        mask = torch.tensor([[0, 0, 1]])
+
+        self._lime_test_assert(
+            func,
+            inp,
+            [[75.0, 75.0, 17.0]],
+            feature_mask=mask,
+            perturbations_per_eval=(1,),
+            target=None,
+            expected_coefs_only=[75.0, 17.0],
+            n_perturb_samples=700,
+        )
+
+    def test_multi_inp_lime_scalar_tensor_0d(self) -> None:
+        net = BasicModel_MultiLayer_MultiInput()
+        self._multi_input_scalar_lime_assert(lambda *inp: torch.sum(net(*inp)))
+
+    def test_multi_inp_lime_scalar_tensor_1d(self) -> None:
+        net = BasicModel_MultiLayer_MultiInput()
+        self._multi_input_scalar_lime_assert(
+            lambda *inp: torch.sum(net(*inp)).reshape(1)
+        )
+
+    def test_multi_inp_lime_scalar_tensor_int(self) -> None:
+        net = BasicModel_MultiLayer_MultiInput()
+        self._multi_input_scalar_lime_assert(
+            lambda *inp: int(torch.sum(net(*inp)).item())
+        )
+
+    def test_multi_inp_lime_scalar_float(self) -> None:
+        net = BasicModel_MultiLayer_MultiInput()
+        self._multi_input_scalar_lime_assert(lambda *inp: torch.sum(net(*inp)).item())
+
+    def _multi_input_scalar_lime_assert(self, func: Callable) -> None:
+        inp1 = torch.tensor([[23.0, 100.0, 0.0], [20.0, 50.0, 30.0]])
+        inp2 = torch.tensor([[20.0, 50.0, 30.0], [0.0, 100.0, 0.0]])
+        inp3 = torch.tensor([[0.0, 100.0, 10.0], [20.0, 10.0, 13.0]])
+        mask1 = torch.tensor([[1, 1, 1]])
+        mask2 = torch.tensor([[0, 1, 2]])
+        mask3 = torch.tensor([[0, 1, 2]])
+        expected = (
+            [[3850.6666, 3850.6666, 3850.6666]],
+            [[305.5, 3850.6666, 410.1]],
+            [[305.5, 3850.6666, 410.1]],
+        )
+
+        self._lime_test_assert(
+            func,
+            (inp1, inp2, inp3),
+            expected,
+            additional_input=(1,),
+            feature_mask=(mask1, mask2, mask3),
+            perturbations_per_eval=(1,),
+            target=None,
+            n_perturb_samples=1500,
+            expected_coefs_only=[305.5, 3850.6666, 410.1],
+            delta=1.5,
+            batch_attr=True,
+        )
+
+    def _lime_test_assert(
+        self,
+        model: Callable,
+        test_input: TensorOrTupleOfTensorsGeneric,
+        expected_attr,
+        expected_coefs_only=None,
+        feature_mask: Union[None, TensorOrTupleOfTensorsGeneric] = None,
+        additional_input: Any = None,
+        perturbations_per_eval: Tuple[int, ...] = (1,),
+        baselines: BaselineType = None,
+        target: Union[None, int] = 0,
+        n_perturb_samples: int = 100,
+        alpha: float = 1.0,
+        delta: float = 1.0,
+        batch_attr: bool = False,
+    ) -> None:
+        for batch_size in perturbations_per_eval:
+            lime = Lime(model, similarity_func=get_similarity_function("cosine", 10.0))
```

Comment:
nit: Are we testing with another similarity function ?

Reply:
Sure, good point, will add another!